shell/platform/linux/fl-dart-project.cc

```diff
@@ -0,0 +1,133 @@
+
+  gchar* assets_path;
+  gchar* icu_data_path;
+};
+
+enum { PROP_ASSETS_PATH = 1, PROP_ICU_DATA_PATH, PROP_LAST };
+
+G_DEFINE_TYPE(FlDartProject, fl_dart_project, G_TYPE_OBJECT)
+
+static void fl_dart_project_set_property(GObject* object,
+                                         guint prop_id,
+                                         const GValue* value,
+                                         GParamSpec* pspec) {
+  FlDartProject* self = FL_DART_PROJECT(object);
+
+  switch (prop_id) {
+    case PROP_ASSETS_PATH:
+      g_free(self->assets_path);
+      self->assets_path = g_strdup(g_value_get_string(value));
+      break;
+    case PROP_ICU_DATA_PATH:
+      g_free(self->icu_data_path);
+      self->icu_data_path = g_strdup(g_value_get_string(value));
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+      break;
+  }
+}
+
+static void fl_dart_project_get_property(GObject* object,
+                                         guint prop_id,
+                                         GValue* value,
+                                         GParamSpec* pspec) {
+  FlDartProject* self = FL_DART_PROJECT(object);
+
+  switch (prop_id) {
+    case PROP_ASSETS_PATH:
+      g_value_set_string(value, self->assets_path);
+      break;
+    case PROP_ICU_DATA_PATH:
+      g_value_set_string(value, self->icu_data_path);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+      break;
+  }
+}
+
+static void fl_dart_project_dispose(GObject* object) {
+  FlDartProject* self = FL_DART_PROJECT(object);
+
+  g_clear_pointer(&self->assets_path, g_free);
+  g_clear_pointer(&self->icu_data_path, g_free);
+
+  G_OBJECT_CLASS(fl_dart_project_parent_class)->dispose(object);
+}
+
+static void fl_dart_project_class_init(FlDartProjectClass* klass) {
+  G_OBJECT_CLASS(klass)->set_property = fl_dart_project_set_property;
+  G_OBJECT_CLASS(klass)->get_property = fl_dart_project_get_property;
+  G_OBJECT_CLASS(klass)->dispose = fl_dart_project_dispose;
+
+  g_object_class_install_property(
+      G_OBJECT_CLASS(klass), PROP_ASSETS_PATH,
+      g_param_spec_string(
+          "assets-path", "assets-path", "Path to Flutter assets", NULL,
+          static_cast<GParamFlags>(G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY |
+                                   G_PARAM_STATIC_STRINGS)));
+  g_object_class_install_property(
+      G_OBJECT_CLASS(klass), PROP_ICU_DATA_PATH,
+      g_param_spec_string(
+          "icu-data-path", "icu-data-path", "Path to ICU data", NULL,
+          static_cast<GParamFlags>(G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY |
+                                   G_PARAM_STATIC_STRINGS)));
+}
+
+static void fl_dart_project_init(FlDartProject* self) {}
+
+/**
+ * fl_dart_project_new:
+ * @assets_path: a file path, e.g. "build/assets"
+ * @icu_data_path: a file path, e.g. "build/icudtl.dat"
+ *
+ * Create a Flutter project.
+ *
+ * Returns: a new #FlDartProject
+ */
```

Comment:
We can move the documentation to the headers. The scanner just parses the contents of the comments, not their locations.

The documentation is in the [Gtk Doc](https://wiki.gnome.org/DocumentationProject/GtkDoc) format. I suspect it wouldn't work as well with Doxygen, as then it would see this as just functions and not objects and methods as they are. The docstrings are also parsed by [GObject introspection](https://gi.readthedocs.io/en/latest/), which is something we might be interested in.

There's an open question of what level of documentation Flutter should provide with the GTK API. It would probably be nice to have documentation, but also it might not be necessary. I suspect many users won't need to modify the generated code, and those who do can probably just read the header files. I think until we decide what documentation to show it's probably better to use the GTK Doc format for now, and reformat later if necessary.